variables.tf

```diff
@@ -68,21 +68,29 @@ variable "drupal_projects_list" {
   }))
 
   validation {
-    # The project name must contain only lower caps letters and "-" and "_" and be between 6 and 16 characters long, since the database user name must be less than 32 chars and we append "<PRJ_ID>_<BRANCH>_dp_u" (i.e.:"_759_main_dp_u") to the project name.
+    # project_name must:
+    # - start with a lowercase letter or a number
+    # - contain only lowercase letters, numbers, hyphens and underscores
+    # and be:
+    # - 6 to 16 characters long if database_host is not null meaning that the database will be created from the module
+    # - 6 to 23 characters long if database_host,database_name,database_user_name are not null meaning that the database already exists
+    # - 6 to 23 characters long if database_host is null meaning that the no database will be created from the module
     condition = alltrue([
       for p in var.drupal_projects_list :
-      can(regex("^[0-9a-z_-]{6,16}$", p.project_name))
+      (can(regex("^[0-9a-z_-]{6,16}$", p.project_name)) && p.database_host != null && p.database_name == null && p.database_user_name == null && can(regex("^[0-9a-z]{1}[0-9a-z_-]+[0-9a-z]{1}$", p.project_name))) ||
+      (can(regex("^[0-9a-z_-]{6,23}$", p.project_name)) && p.database_host != null && p.database_name != null && p.database_user_name != null && can(regex("^[0-9a-z]{1}[0-9a-z_-]+[0-9a-z]{1}$", p.project_name))) ||
+      (can(regex("^[0-9a-z_-]{6,23}$", p.project_name)) && p.database_host == null && can(regex("^[0-9a-z]{1}[0-9a-z_-]+[0-9a-z]{1}$", p.project_name)))
```

Comment:
Can we aggregate the p.project_name regex using a single one? For example:

`"^[0-9a-z]{1}[0-9a-z_-]{4,14}[0-9a-z]{1}$"`

Can you please test it?